R/dataPrepRast.R

```diff
@@ -0,0 +1,131 @@
+#' projection(inputrast) <- CRS("+proj=longlat +datum=WGS84")
+#'
+#' #Extract Canada boundary from wrld_simpl
+#' canada<-wrld_simpl[which(wrld_simpl@data$NAME =='Canada'),]
+#' #dataPrepRast
+#' x<-dataPrepRast(x=inputrast, cropWith=canada , maskWith= canada,
+#'                  reprojectTo= inputrast, timings = TRUE)
+dataPrepRast <- function(x, cropWith=NULL, maskWith=NULL, reprojectTo=NULL, outPath=NULL,
+                   fileOut=NULL, formatOut=NULL, timings=FALSE, ...) {
+  wd<-getwd()
+  if(!is.null(outPath)) setwd(outPath)
+  if(!is.null(cropWith)) cropWithname <- deparse(substitute(cropWith))
+  if(!is.null(maskWith)) maskWithname <- deparse(substitute(maskWith))
+  xname <- deparse(substitute(x))
+
+  # Set input list with names
+  inputls<-list(x, cropWith, maskWith, reprojectTo)
+  name<- c("x","cropWith","maskWith","reprojectTo")
+  names(inputls)<-name
+
+  # remove NULL
+  ilst<-inputls[!sapply(inputls, is.null)]
+
+  # set output projection. If not defined, derive reprojectTo using crs of param x
+  if(is.null(reprojectTo)) {
+    if(!is.na(crs(ilst$x))){
+      reprojectTo <- crs(ilst$x)
+    }else{
+      stop(paste("You must provide a coordinate system (reprojectTo) or",
+            "have your input object assigned to a coordinate reference"))
+    }
+  }else{
+    reprojectTo<-crs(ilst$reprojectTo)
+  }
+
+  # reproject all inputs into the same CRS
+  ilst<- lapply(ilst, function(i) {
+    if (is(i, "SpatialPolygonsDataFrame")){
+      if(!compareCRS(i,reprojectTo))   i<- spTransform(i, reprojectTo)
+    } else if (is(i, "RasterLayer")){
+      if (!compareCRS(i,reprojectTo))  i<- projectRaster(i,crs=reprojectTo)
+    }else {
+      stop("object class is not recognized")
+    }
+    return(i)
+  })
+
+  #crop
+  if (!is.null(cropWith)){
+    if (timings) preTime <- Sys.time()
+    # Test if x and maskWith intersect.
+    # To avoid empty output if extents do not intersect while using rasterize()
+    doIntersect<- isIntersect(ilst$x, ilst$cropWith)
+    if (doIntersect) {
+      ilst$x<-crop(ilst$x,ilst$cropWith)
+      if (timings) {
+        msg <- capture.output(
+          print(paste("finished croping ", xname, "with", cropWithname)),
+          print(paste("took", format(Sys.time()-preTime, digits=2)))
+        )
+        message(msg)
+      }
+    }else{
+      message("Crop did not occur")
+    }
+
+  }
+
+  #mask
+  if(!is.null(maskWith)){
+    if (timings) preTime <- Sys.time()
+    # Test if x and maskWith intersect.
+    # To avoid empty output if extents do not intersect while using rasterize()
+    doIntersect<- isIntersect(ilst$x, ilst$maskWith)
+    if(doIntersect){
+      ilst$x <- mask(ilst$x, mask=ilst$maskWith)
+      if (timings) {
+        msg <- capture.output(
+          print(paste("finished masking ", xname, "with", maskWithname)),
+          print(paste("took", format(Sys.time()-preTime, digits=2)))
+        )
+        message(msg)
+      }
+    }else{
+         message("Mask did not occur.")
```

Comment:
Why not a warning/error? If the user wants to mask and the areas don't overlap, then that should be an error.

Also, the message needs to be more explicit about why the mask didn't occur (*e.g.*, "areas do not intersect")